### PR TITLE
Add posibility for default_timeout

### PR DESCRIPTION
**Pull request by sbv-csis**

### CODE REVIEW

**Comment by jea-csis** [comment]
LGTM :+1:

File: src/proccer/agent.py [hunk 1]
```diff
@@ -48,6 +50,12 @@ def read_configuration(fp):
             if m.match(job_id):
                 desc.update(override)
 
+    if 'default-timeout' in parsed:
+        default_timeout = parse_interval(parsed.pop('default-timeout'))
+        default_timeout = int(default_timeout.total_seconds())
+        for name, desc in parsed['commands'].items():
+            desc['timeout'] = desc.get('timeout', default_timeout)
```
Comment (ski-csis) [inline src/proccer/agent.py]: You `parse_interval` the default, should you not `parse_interval` the per-job one too?

File: src/proccer/agent.py [hunk 2]
```diff
@@ -48,6 +50,12 @@ def read_configuration(fp):
             if m.match(job_id):
                 desc.update(override)
 
+    if 'default-timeout' in parsed:
+        default_timeout = parse_interval(parsed.pop('default-timeout'))
+        default_timeout = int(default_timeout.total_seconds())
+        for name, desc in parsed['commands'].items():
```
Comment (ski-csis) [inline src/proccer/agent.py]: Is there a reason you use `.items()` and ignore the keys? `.values()` is probably more descriptive...

File: src/proccer/t/test_agent.py [hunk 3]
```diff
@@ -72,6 +72,34 @@ def test_with_overrides():
...
 
+def test_with_default_timeout():
+    yaml = '''
+        default-timeout: 5 seconds
+        commands:
+            noto:   {command: "true", warn-after: 15 seconds}
+            withto:
+                command: "true"
+                warn-after: 15 seconds
+                timeout: 12345
```
Comment (ski-csis) [inline src/proccer/t/test_agent.py]: The test does not test that you parse per-job timeouts :)

**Comment by ski-csis** [comment]
Nice :+1:

**Comment by ski-csis** [comment]
:+1:

**Pull request merged**